standard/lpdb.lua

```diff
@@ -69,4 +77,136 @@ function Lpdb.executeMassQuery(tableName, queryParameters, itemChecker, limit)
 	end
 end
 
+--- LPDB Object-Relational Mapping
+
+---@alias ModelColumnData {name: string, fieldType: string|any, default: any}
+
+---@class Model
+---@field tableName string
+---@field tableColumns ModelColumnData[]
+local Model = Class.new(function(self, name, columns)
+	self.tableName = name
+	self.tableColumns = columns
+end)
+
+---@class ModelRow
+---@field private tableName string
+---@field private tableColumns ModelColumnData[]
+---@field private fields table<string, any>
+---@field [any] any
+local ModelRow = Class.new(function(self, tableName, tableColumns)
+	rawset(self, 'tableName', tableName)
+	rawset(self, 'tableColumns', tableColumns)
+	rawset(self, 'fields', {})
+end)
+
+---@param initData table?
+---@return ModelRow
+function Model:new(initData)
+	local row = ModelRow(self.tableName, self.tableColumns)
+	if type(initData) == 'table' then
+		row:setMany(initData)
+	end
+	return row
+end
+
+---@private
+---@param columnData ModelColumnData
+function ModelRow:_validateField(columnData)
+	if not self.fields[columnData.name] then
+		error(self.tableName .. ' expects ' .. columnData.name .. ' to be set')
+	end
+	-- TODO: Verify types (at least when running tests)
+end
+
+---@private
+---@param columnData ModelColumnData
+function ModelRow:_prepareFieldForStorage(columnData)
+	-- Apply defaults
+	if not self.fields[columnData.name] then
+		if type(columnData.default) == 'function' then
+			self.fields[columnData.name] = columnData.default(self.fields)
+		else
+			self.fields[columnData.name] = columnData.default
+		end
+	end
+
+	-- Validate that all fields are correct
+	self:_validateField(columnData)
+end
+
+function ModelRow:save()
+	Array.forEach(self.tableColumns, FnUtil.curry(ModelRow._prepareFieldForStorage, self))
+	local objectName = Table.extract(self.fields, 'objectname')
+	mw.ext.LiquipediaDB['lpdb_' .. self.tableName](objectName, self.fields)
+end
+
+function ModelRow:__newindex(key, value)
+	-- Strip HTML from strings
```

Comment:
what if we have stuff that actually is supposed to be stored with html?
(e.g. inside a comment)

Reply:
Not sure the effect of it is good though. I can see it being allowed in extradata, but probably want to keep standard fields free of HTML

Reply:
i can only think of usage in extradata atm, so in extradata would be fine